crates/fs/public/node.rs

```diff
@@ -1,36 +1,37 @@
-//! Public file system in-memory representation.
+//! Public node system in-memory representation.
 
-use std::{
-    io::{Cursor, Read, Seek},
-    rc::Rc,
-    result,
-};
+use std::rc::Rc;
 
 use anyhow::{bail, Result};
+use async_trait::async_trait;
 use chrono::{DateTime, Utc};
-use libipld::{cbor::DagCborCodec, codec::Decode, Cid};
+use libipld::{Cid, Ipld};
+use serde::{de, Deserialize, Deserializer, Serializer};
 
-use super::{Id, PublicDirectory, PublicFile};
-use crate::{common::BlockStore, FsError, UnixFsNodeKind};
+use super::{PublicDirectory, PublicFile};
+use crate::{common::BlockStore, AsyncSerialize, FsError, Id, Metadata, UnixFsNodeKind};
+
+//--------------------------------------------------------------------------------------------------
+// Type Definitions
+//--------------------------------------------------------------------------------------------------
 
 /// A node in a WNFS public file system. This can either be a file or a directory.
-#[derive(Debug, Clone, PartialEq, Eq)]
+///
+/// PublicNode is serialized and deserialized as [untagged][1] enum.
+///
+/// [1]: https://serde.rs/enum-representations.html#untagged
```

Comment:
Again, I know, WIP, but just making sure we don't forget to remove this :)

Reply:
I need to reword this because it is still serialized untagged